src/main/scala/it/pagopa/interop/partymanagement/api/impl/PublicApiServiceImpl.scala

```diff
@@ -305,9 +305,11 @@ class PublicApiServiceImpl(
   private def isTokenNotConsumed(tokenId: String, relationships: Seq[PersistedPartyRelationship]): Future[Unit] = {
     val error: Either[Throwable, Unit] = Left(TokenAlreadyConsumed(tokenId))
     error
-      .unlessA(relationships.nonEmpty && relationships.forall(_.state == Pending))
+      .unlessA(
+        relationships.nonEmpty &&
+          (relationships.forall(_.state == Pending) || relationships.forall(_.state == ToBeValidated))
```

Comment:
Are you sure that the check has to be this? If `ToBeValidated` is an accepted state like `Pending` hasn't the check to be something like this?

```suggestion
          (relationships.forall(s => s.state == Pending || s.state == ToBeValidated)
```

Reply:
updated